addons/mail/static/src/models/messaging_notification_handler.js

```diff
@@ -452,12 +432,10 @@ registerModel({
         /**
          * @private
          * @param {Object} data
-         * @param {Object} [data.volumeSettings]
+         * @param {Object} [data.volume_settings_ids]
          */
-        async _handleNotificationVolumeSettingUpdate({ volumeSettings }) {
-            this.messaging && this.messaging.userSetting.update({
-                volumeSettings: volumeSettings,
-            });
+        async _handleNotificationVolumeSettingUpdate({ volume_settings_ids }) {
+            this.messaging.currentUser.resUsersSettings.update({ volume_settings_ids });
```

Comment:
param just data, and update data?
Also this should ideally become just insert on `res.users.settings.volumes` with id provided to identify it

and notif type just `res.users.settings.volumes/insert`